rust/flexbuffers/src/flexbuffer.rs

```diff
@@ -0,0 +1,89 @@
+use std::ops::{RangeBounds, Rem, Bound};
+use std::slice::SliceIndex;
+use std::fmt::Debug;
+
+use crate::reader::ReadLE;
+
+pub trait FlexBuffer: Debug + PartialEq + IntoIterator + AsRef<[u8]> + Default + Clone {
```

Comment:
nit: I think `FlexBuffer` is a confusing name and it should be called just `Buffer` or `FlexbufferBacking` or something else.

Reply:
How about `InternalBuffer`? Otherwise just `Buffer` works.